ncm-network/src/main/perl/nmstate.pm

```diff
@@ -103,6 +104,26 @@ sub disable_nm_manage_dns
     };
 }
 
+sub nm_create_device_config_dropin
+{
+    my ($self, $nm_device_config, $nwsrv) = @_;
+    my @data = ('[device]');
+
+    if ( $nm_device_config ) {
+        foreach my $key (sort keys %$nm_device_config){
+            push @data, $key."=".$nm_device_config->{$key};
+        };
+
+        $self->verbose("setting device configuration dropin");
+    }
+    my $fh = CAF::FileWriter->new($NM_DEVICE_DROPIN_CFG_FILE, mode => oct(444), log => $self, keeps_state => 1);
```

Comment:
keeps_state seems wrong here if a successful write results in a network reload?

Reply:
I don't think I fully understand what keeps_state purpose is?

Reply:
It means that writing the file doesn't affect the state of the system, which isn't the case here?

> **keeps_state**
>
> A boolean specifying whether a file change respects the current system state or not. A file with keeps_state will be created/modified, regardless of any value for NoAction. This is useful when creating temporary files that are required for a NoAction run.
> By default, file changes modify the state and thus keeps_state is false.

https://quattor-documentation.readthedocs.io/en/stable/CAF/CAF_FileWriter.html#public-methods

Reply:
ah thanks, will remove that then if default is false, also may need to fix it in sub disable_nm_manage_dns() too.